### ISSUE TITLE
Navigation: collected material layout

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#1868]

**Comment by urapadmin** [comment]
- test jump to trench (footsteps)
- test jump to locus (footsteps)
- back button and history
- arrow keys and selecting a different cm from the list should make an entry in the recently viewed list
- add a new cm should add that in the recently viewed list

**Comment by luizaogs** [comment]
The new cm does appear in the recently viewed list, but the material in parenthesis does not: 

<img width="1025" alt="Screen Shot 2022-10-19 at 9 31 08 PM" src="https://user-images.githubusercontent.com/67970402/196842264-0a57a792-f189-4f4d-845f-a9d333c97976.png">

**Comment by luizaogs** [comment]
Ah, if I leave that layout and then come back to the CM it then appears, but it appears twice, one with the material and one without: 

<img width="1024" alt="Screen Shot 2022-10-19 at 9 32 31 PM" src="https://user-images.githubusercontent.com/67970402/196842460-e7cadf13-83e1-498c-817a-9962d7924aa4.png">

**Comment by luizaogs** [comment]
Though actually hm you didn't put the release above so I don't know if I'm supposed to be testing this yet. If not, my bad!

**Comment by urapadmin** [comment]
I forgot the release, sorry. This was ready for testing.

I thought I tackled this. Is this adding related to adding a new cm within the layout or so?

**Comment by luizaogs** [comment]
Yes, I added it using the plus sign between the arrows in the header.

**Comment by urapadmin** [comment]
I can't remember how PVD is set up. Do you have to select the material when you create the collected material or do you select it afterwards in the material selection on the cm layout? Which would indeed change the line after its creation.

**Comment by luizaogs** [comment]
I select it in the cm layout. 

And I just tried this again and the material is only added to the recently viewed, it seems, after I leave the CM layout itself and come back. Simply moving between CM doesn't add it to the recently viewed. In case that helps.

**Comment by urapadmin** [comment]
hm. that makes it even weirder. Seems I have to try harder on that one ... tomorrow. 🛌

**Comment by luizaogs** [comment]
Go to bed :)

**Comment by urapadmin** [comment]
you too! 💤

**Comment by urapadmin** [comment]
I merge this here and
https://github.com/arch-kiosk/arch-kiosk-office/issues/1862#issuecomment-1286237419
So please go look there. I close this here as a duplicate.

**Comment by urapadmin** [comment]
aaah wait. Beside that issue navigation was okay here, right?

**Comment by urapadmin** [comment]
last changes because of #1862 requires a new test of this one here, anyhow.

**Comment by luizaogs** [comment]
Yes, everything else was fine, but I shall check again with the new release.

**Comment by urapadmin** [comment]
### recording 13.6.6
Ja, I forgot the new release number again. hmpf.

**Comment by luizaogs** [comment]
Argh ok I know I closed #1862 but I now have a question about something I hadn’t noticed before so I will ask here since it is a duplicate. When adding a new cm, the material doesn’t show up in the navigation history. We know that. It DOES show up in parenthesis if I come back to the cm page in any way other than through the navigation history. If I go back to the cm from there the material does NOT update in the history. Is that ok?

**Comment by luizaogs** [comment]
I don't think I explained it properly. Hmpf let me try it again.

**Comment by luizaogs** [comment]
Steps: 
1. Create new cm in the cm layout (LA-12). Give it the material glass. In the recently viewed, I get LA-12 sample().
2. Navigate to LA-11 (ivory) for funsies. Click on recently viewed: LA-12 still shows up as LA-12 sample().
3. I go back to LA-12 through the recently viewed. After getting back there, the recently viewed looks like this: 
![IMG_0316](https://user-images.githubusercontent.com/67970402/197088773-f232c564-adbe-441c-8aec-60d6284c5da8.PNG)

4. I then navigate to LA-10 through the cm list in the cm layout.
5. Navigate back to LA-12 but this time using the arrows.
6. Now and only now do I get this in the recently viewed, LA-12 with its material: 
![IMG_0317](https://user-images.githubusercontent.com/67970402/197088792-6c576d38-4225-47b2-924f-480e0ee7df7f.PNG)

So my question basically is, is this sequence of events ok? Or should (glass) have showed up in the recently viewed next to LA-12 after I navigated back to that record through the recently viewed list itself? (See step 3)

**Comment by luizaogs** [comment]
(So oops it wasn't navigation history in the first comment, it should have been recently viewed.)

**Comment by urapadmin** [comment]
ja, that is the sort of intended behavior. It does not change by changing the material. It will be replaced next time you navigate to the same record again by whatever way. So leaving the layout and going back to it would have done the same. But leaving the layout alone does not. Still not perfect but what ever is?

**Comment by urapadmin** [comment]
ah no, wait. I read it again. What isn't entirely clear is why it does not get replaced with the correct entry when you navigate from LA-11 to LA-12 by means of the recently viewed. I will look at that one again.

**Comment by luizaogs** [comment]
Yup, that's it. Sorry, complicated!

**Comment by urapadmin** [comment]
I knew why I would not want to touch this topic for years :o) It can get nightmarish.
But I got this one.

### recording 13.6.7
- got ya 🐛

**Comment by luizaogs** [comment]
Argh Lutz, I'm sorry -- now when I add a new cm in the cm layout it does not show up in the recently viewed at all. Only if I come back to it after moving to another cm.

(The bug you said you caught above you did catch.)

**Comment by urapadmin** [comment]
### recording 13.6.8
- I have added a line that should fix that (also for loci, where it could have easily been the same issue).

**Comment by luizaogs** [comment]
Ok I THINK this is ok now. But I just realized that I manage to get two entries in layout history, one with the material and one without. I think this is what you said might sometimes happen in #1862 so I think it's ok, but am checking with you before the ticket gets closed: 

<img width="1023" alt="Screen Shot 2022-10-23 at 11 37 43 PM" src="https://user-images.githubusercontent.com/67970402/197449564-8ee1af67-82ed-4b3f-8f53-f0fad91c089c.png">

**Comment by urapadmin** [comment]
yes, in the layout history that can happen. I don't "rename" all existing entities in the layout history because it would take too long to catch that rarity. So yes, this is a case of #1862. Clearly no Covid 🧠 on your side!
